File: strategy/java/MallardDuck.java
package strategy.java;

import strategy.java.fly.FlyWithWings;
import strategy.java.quack.Quack;

public class MallardDuck extends Duck {
    public MallardDuck() {
        this.flyBehavior = new FlyWithWings();
        this.quackBehavior = new Quack();
    }

    public void display() {
        
    }
}

File: strategy/typescript/classes/ducks/mallard-duck.ts
import { Duck } from "../duck";
import { FlyWithWings } from "../flyBehaviors/fly-with-wings";
import { Quack } from "../quackBehaviors/quack";

export class MallardDuck extends Duck {
  public constructor() {
    super();

    this.quackBehavior = new Quack();
    this.flyBehavior = new FlyWithWings();
  }

  display(): void {
    console.log("Im look like a mallard duck");
  }
}
